First shot at a revamped documentation (#310)
CanalTP/mimirsbrunn: README.md

[![GitHub license](https://img.shields.io/github/license/CanalTP/mimirsbrunn.svg)](https://github.com/CanalTP/mimirsbrunn/blob/master/LICENSE)
[![GitHub tag](https://img.shields.io/github/tag/CanalTP/mimirsbrunn.svg)](https://github.com/CanalTP/mimirsbrunn/tag)

# Mimirsbrunn

Mimirsbrunn (also called Mimir) is an independent geocoding and reverse-geocoding system written in [Rust](https://www.rust-lang.org/en-US/), and built upon [Elasticsearch](https://www.elastic.co).
It can handle addresses, streets, points-of-interest (POI), administrative regions or public transport stops.

## what's a geocoder ?

> For [Wikipedia](https://en.wikipedia.org/wiki/Geocoding), a geocoder is «the computational process of transforming a physical address description to a location on the Earth's surface».

Reviewer: ```suggestion
Usually [geocoding](https://en.wikipedia.org/wiki/Geocoding) refers to "the process of transforming a physical address description to a location on the Earth's surface". 
However Mimir is more a [geoparser](https://en.wikipedia.org/wiki/Toponym_resolution#Geoparsing) than a geocoder since it can resolve any ambiguous toponym to its correct location.
```

In other words, a geocoder reads a description (possibly incomplete) of a location, and returns a list of candidate locations (latitude / longitude) matching the input. 

> Geocoding is traditionally used for autocompleting search fields used in geographic applications. For example, here is a screenshot of Qwant Maps, where the user enters a search string 20 rue hec mal, and mimir returns possible candidates in a dropdown box.

Reviewer: Mimir is mainly based on several heuristics and hand-crafted rules upon Elasticsearch.

Author: hum while it is indeed interesting to say this somewhere, I don't think that fits well in this section, I think it is too technical.

Reviewer: Indeed.

> ![qwnat maps](https://user-images.githubusercontent.com/3987698/56976025-53ed1180-6b72-11e9-9c81-9718e92061ce.png)

Reviewer 2: ```suggestion
![qwant maps](https://user-images.githubusercontent.com/3987698/56976025-53ed1180-6b72-11e9-9c81-9718e92061ce.png)
```

## who uses it ?

* [Navitia](https://github.com/CanalTP/navitia)
* [Qwant Maps](https://www.qwant.com/maps)

> If you uses it too, feel free to open a pull request, we'll be happy to add your project here!

Reviewer: If you use... (without "s")

[![GitHub license](https://img.shields.io/github/license/CanalTP/mimirsbrunn.svg)](https://github.com/CanalTP/mimirsbrunn/blob/master/LICENSE)
[![GitHub tag](https://img.shields.io/github/tag/CanalTP/mimirsbrunn.svg)](https://github.com/CanalTP/mimirsbrunn/tag)

# Mimirsbrunn

> Mimirsbrunn (also called Mimir) is an independent geocoding and reverse-geocoding system written in [Rust](https://www.rust-lang.org/en-US/) and built upon [Elasticsearch](https://www.elastic.co).

Reviewer 3: ```suggestion
Mimirsbrunn (also called Mimir) is an independent geocoding and reverse-geocoding system written in [Rust](https://www.rust-lang.org/en-US/), and built upon [Elasticsearch](https://www.elastic.co).
```

It can handle addresses, streets, points-of-interest (POI), administrative regions or public transport stops.

## what's a geocoder ?

For [Wikipedia](https://en.wikipedia.org/wiki/Geocoding), a geocoder is «the computational process of transforming a physical address description to a location on the Earth's surface».

> A more mumdane way of explaing it would be to say that a geocoder takes as input some text (like "20 rue hector malot paris", or "central park new y") and returns some likey several candidates latitude/longitude location for this place.

Reviewer 3: ```suggestion

In other words, a geocoder reads a description (possibly incomplete) of a location, and returns a list of candidate locations (latitude / longitude) matching the input. 
```

Author: the rephrasing is indeed way better, but don't you think adding example helps the understanding ? maybe after this sentence ?

Reviewer 3: Yes, the example adds clarity.... and comes right after in terms of a screenshot

> It is used for example in most map services as the search field:

Reviewer 3: Geocoding is traditionally used for autocompleting search fields used in geographic (oriented?) applications. For example, here is a screenshot of [Qwant Maps](https://www.qwant.com/maps), where the user enters a search string `20 rue hec mal`, and `mimir` returns possible candidates in a dropdown box.

![qwnat maps](https://user-images.githubusercontent.com/3987698/56976025-53ed1180-6b72-11e9-9c81-9718e92061ce.png)

## who uses it ?
[Navitia](https://github.com/CanalTP/navitia) and [Qwant Maps](https://www.qwant.com/maps) use it as their global geocoding service.
If you uses it too, feel free to open a pull request, we'll be happy to add your project here!

## ressources

* [A french presentation of Mimirsbrunn](https://github.com/TeXitoi/pinot2017bano/blob/master/pinot2017bano.pdf)

# How to use

## API

Mimirsbrunn exposes a [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) Json api (with [bragi](https://github.com/CanalTP/mimirsbrunn/tree/master/libs/bragi)).

This API provices several services:

> ### Autocomplete

Reviewer 3: Maybe format as a table
